code/obj/item/grenades.dm

```diff
@@ -919,24 +920,37 @@ PIPE BOMBS + CONSTRUCTION
 				boutput(user, "<span class='alert'>Huh? How does this thing work?!</span>")
 				src.primed = TRUE
 				SPAWN_DBG( 5 )
-					boom()
+					boom(user)
 					return
 
 			else
 				boutput(user, "<span class='alert'>You prime the firework! [det_time/10] seconds!</span>")
 				src.primed = TRUE
 				SPAWN_DBG( src.det_time )
-					boom()
+					boom(user)
 					return
 
-	proc/boom()
+	proc/boom(mob/user as mob)
 		var/turf/location = get_turf(src.loc)
+		if (!ishuman(user))
+			return
```

Comment:
You don't want to return, because that means the firework won't explode at all if you're not human. Instead, you should just do:

```
if(ishuman(user))
    cast to human
    do all the human related limb stuff
```

Reply:
moved ishuman, casting, and limb stuff into and under bootleg_level check